internal/rbd/rbd_util.go

```diff
@@ -335,38 +331,62 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 	image := pOpts.RbdImageName
 	found, _, err := rbdStatus(ctx, pOpts, cr)
 	if err != nil {
+		klog.Errorf(util.Log(ctx, "failed getting information for image (%s): (%s)"), pOpts, err)
+		if strings.Contains(err.Error(), "rbd: error opening image "+image+
+			": (2) No such file or directory") {
+			return ErrImageNotFound{image, err}
+		}
 		return err
 	}
 	if found {
-		klog.Errorf(util.Log(ctx, "rbd %s is still being used"), pOpts)
-		return fmt.Errorf("rbd %s is still being used", pOpts)
+		klog.Errorf(util.Log(ctx, "rbd is still being used "), image)
+		return fmt.Errorf("rbd %s is still being used", image)
 	}
+	// Support deleting the older rbd images whose imageID is not stored in omap
+	if pOpts.ImageID == "" {
```

Comment:
the `pOpts.ImageID == ""` check should be inside `pOpts.getImageID()`, so that you can always call the function.